lms/resources/_js_config.py

```diff
@@ -363,7 +364,33 @@ def _hypothesis_client(self):
                     "authority": self._authority,
                     "enableShareLinks": False,
                     "grantToken": self._grant_token(api_url),
-                    "groups": [self._context.h_group.groupid(self._authority)],
+                    "groups": self._groups(),
                 }
             ]
         }
+
+    def _groups(self):
+        if self._context.should_use_section_groups:
+            return "$rpc:requestGroups"
+        return [self._context.h_group.groupid(self._authority)]
+
+    def _sync_api(self):
+        if not self._context.should_use_section_groups:
+            return None
+
+        req = self._request
+
+        return {
+            "path": req.route_path("api.sync"),
+            "data": {
+                "lms": {
+                    "tool_consumer_instance_guid": req.params[
+                        "tool_consumer_instance_guid"
+                    ],
+                },
+                "course": {
+                    "context_id": req.params["context_id"],
+                    "custom_canvas_course_id": req.params["custom_canvas_course_id"],
+                },
+            },
```

Comment:
These are the params that the sync API needs, in its request body

Reply:
Because of the way [the frontend's API client](https://github.com/hypothesis/lms/blob/master/lms/static/scripts/frontend_apps/utils/api.js) works, it has to be in this format: it doesn't accept a `method` argument but instead [sets the method depending on whether there's a body or not](https://github.com/hypothesis/lms/blob/809b24db432e8bd61ebd294b06e7c5372b5e0456/lms/static/scripts/frontend_apps/utils/api.js#L59) and it [assumes the body is an object not a string](https://github.com/hypothesis/lms/blob/809b24db432e8bd61ebd294b06e7c5372b5e0456/lms/static/scripts/frontend_apps/utils/api.js#L54)

Reply:
I think there's some work to be done on the frontend here:

* Not all API requests work by the backend passing all the request details in the JS config like this. In most cases the frontend constructs the URL and params itself. I think they should all be changed to work this way: the backend passing the request deets and the frontend just being dumb

* The backend could also stringify the body and set the method and the headers. The frontend could be dumb about all that too

Should result in overall simpler frontend code.

For another time maybe

Reply:
Yeah, I didn't know if this was using a new pathway written for this, but if it's existing stuff, so be it. I do think it would be nice in future, but it's not exactly burning the house down.